src/Extensions/TimedScopes/TaskExtensions.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Threading.Tasks;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.TimedScopes
+{
+	/// <summary>
+	/// Extensions for Task to wrap them in TimedScope
+	/// </summary>
+	public static class TaskExtensions
+	{
+		/// <summary>
+		/// Starts TimedScopes before task execution and finish after it's completion
+		/// </summary>
+		public static async ValueTask<T> WithTimedScope<T>(this ValueTask<T> task, ITimedScopeProvider provider, TimedScopeDefinition definition)
+		{
+			using TimedScope timedScope = provider.Start(definition, TimedScopeResult.SystemError);
+			T result = await task.ConfigureAwait(false);
+			timedScope.Result = TimedScopeResult.Success;
+			return result;
+		}
+
+
+		/// <summary>
+		/// Starts TimedScopes before task execution and finish after it's completion
+		/// </summary>
+		public static async ValueTask WithTimedScope(this ValueTask task, ITimedScopeProvider provider, TimedScopeDefinition definition) =>
+			await ConvertToTaskWithResultValue(task).WithTimedScope(provider, definition).ConfigureAwait(false);
```

Comment:
Remove `async` and `await` here

Reply:
It's added there because there is no way to convert from `ValueTask<T>` to `ValueTask` from what I was able to understand (discussion about it here https://github.com/dotnet/runtime/issues/31503), please let me know if there is a way.

Reply:
You can do something like:
```
    if (valueTask.IsCompletedSuccessfully)
    {
        valueTask.GetResult();
        return default;
    }

    return new ValueTask(valueTask.AsTask());
```

Reply:
I've seen this suggestion, but I'm not sure that I want to use it, because of a call to `AsTask()` that will create a Task and defeats the purpose of having ValueTask.
Do you think having a state machine is worth then creating a Task object? Or I misunderstanding something about how it will work?

Reply:
After some discussion, we've decided to leave it like this since AsTask would probably allocate an object versus state machine that will be a struct in release build https://sharplab.io/#v2:D4AQDABCCMCsDcBYAUCkAmCBhCBvFEhUAHBAGoCGANgK4CmIAbBACIAUAlHgUbyAJxRGAOhZ0qFAJ5toHJMl4BfFIqA=

Reply:
Oh, sorry I didn't see that. If you're doing additional pushes then maybe just put in a short comment on that :).

Reply:
Will do, thanks